common/djangoapps/student/views/dashboard.py

```diff
@@ -680,6 +680,11 @@ def student_dashboard(request):  # lint-amnesty, pylint: disable=too-many-statem
         for enrollment in course_enrollments
     }
 
+    passing_courses = {
+        enrollment.course_id: user_has_passing_grade_in_course(user, enrollment)
+        for enrollment in course_enrollments
```

Comment:
Are we worried about the performance if the user has a lot of enrollments? Will this slow the dashboard down?

Reply:
I took two (semi) related loops and smushed them together.